### ISSUE TITLE
Android Studio / xtend gradle issue due to trace files in generated source directory

### ISSUE THREAD

**Issue by bitcubate** [xtext/xtend-gradle-plugin#30]
Hi I get this error in Android Studio:

IO error while decoding /media/matrix/XDEV/dev-android/as_workspace/BitCubateXCore/app/build/generated/source/xtend/debug/bc/core/x/.ActCoreXtend.java._trace with UTF-8
Please try specifying another one using the -encoding option

The trace file seems to be binary and I can't get rid of it. 

Here are my .build file contents:

buildscript {
    repositories {
        mavenCentral()
 		jcenter()
    }
    dependencies {
        classpath "com.android.tools.build:gradle:1.1.3"
        classpath "org.xtend:xtend-android-gradle-plugin:0.4.7"
      }
}

and in the app build file I have this compile dependency:

    compile 'org.eclipse.xtext:org.eclipse.xtext.xbase.lib.slim:2.8.2'

also plugin is applied:

apply plugin: "org.xtend.xtend-android"

Could this be a bug?

Are there any workarounds?

Best Regards
Robert N

**Comment by oehme** [comment]
Xtend needs the .trace files to later install debug information into the .class files. I don't know why and where AS is trying to parse them. Could you give me the full error message?

**Comment by bitcubate** [comment]
Hi Stefan

Ignore my previous issue, I figured out that the error is due to having a Android Scala and Xtend mixed project. The Scala Zinc compiler seems to compile xtend generated java files and fails on those trace files. It's not your plugin's fault.

**Comment by oehme** [comment]
Thanks for reporting back and your encouraging feedback =)

I guess you should file a bug with the Scala plugin. They shouldn't just pass all files to the compiler, that's a little extreme.

**Comment by bitcubate** [comment]
Hi Stefan

Thanks for the suggestion, I will raise an issue/feature request for the scala android plugin. Figured out that Zinc does incremental compile for the whole project, hence fails. Solution was to add this to build.gradle so that it would compile only scala and in a separate ant task.

tasks.withType(ScalaCompile) {
    scalaCompileOptions.useAnt = true
}

Also here is a demo of 7 JVM languages in a single demo app: https://youtu.be/wNa-tGBuaLQ . It's a stupid demo, but contains Scala, Groovy, Beanshell, Lua, Xtend, Kotlin and Processing. Right now I used standard example code for the activities from users at Github - will need to compile their list and credit them before release.

Will publish to github once I make proper activities in each language so that it can be licensed as MIT and to serve as a Android Rosetta Stone for people who want to code in something (easier) besides Java. If you or anyone else you know would like to add awesome Xtend activities, feel free.

Thanks again for developers like you who make tools that make other peoples life easy! Tool developers are unsung heroes of the development world - nobody really knows them.

Best Regards
Robert N
